js/timers.js

```diff
@@ -6,61 +6,36 @@ function openPanel() {
         elements[i].classList.remove("hidden");
     }
     document.getElementById("addtimerbutton").classList.remove("hidden");
-
-    var timerelements = document.getElementsByClassName("timer");
-    Array.prototype.forEach.call(timerelements, function(ele){
-                ele.dataset.state="settings";
-        }
-    );
     document.getElementById("timerConfig").dataset.state = "open";
     document.getElementById("timerConfig").classList.add("timerConfigAnimateIn");
     document.getElementById("timerConfig").classList.remove("timerConfigAnimateOut");
-    document.getElementById("timers").dataset.state = "settings";
-    var timers = document.getElementById("timers").children;
-    for (i=0; i<timers.length; ++i) {
-        timers[i].dataset.state = "settings";
-    }
 }
 
 function closePanel() {
-    document.getElementById("timers").dataset.state = "";
-    document.getElementById("timerConfig").dataset.state = "closed";
+    var timers = document.querySelectorAll(".timer");
+    Array.prototype.forEach.call(timers, function(item){
+        item.dataset.state="un-selected";
+    });
     document.getElementById("timerConfig").classList.add("timerConfigAnimateOut");
     document.getElementById("timerConfig").classList.remove("timerConfigAnimateIn");
 }
 
 function hideConfigElements() {
-    // hide the internal elements after closing the panel since it looks better that way
-    if (document.getElementById("timerConfig").dataset.state == "closed") {
-        var timers = document.getElementById("timers").children;
-        for (i=0; i<timers.length; ++i) {
-            if (timers[i].dataset.state == "settings" || timers[i].dataset.state == "settings-selected") {
-                timers[i].dataset.state = "closed";
-            }
-        }
```

Comment:
Its stuff like this that worry me. Is there a reason that you changed the timers class so much? I don't know what it is doing well enough to feel good about approving these changes.